bears/general/FilenameBear.py

```diff
@@ -71,15 +81,32 @@ def run(self, filename, file,
                      file_naming_convention)
             return
 
+        if new_name != filename_without_extension:
+            messages.append(
+                'Filename does not follow {} naming-convention.'.format(
+                    file_naming_convention))
+
+        if not filename_without_extension.startswith(filename_prefix):
+            new_name = filename_prefix + new_name
+            messages.append(
+                'Filename does not use the {!r} prefix.'.format(
+                    filename_prefix))
+
+        if not filename_without_extension.endswith(filename_suffix):
+            new_name = new_name + filename_suffix
+            messages.append(
+                'Filename does not use the {!r} suffix.'.format(
+                        filename_suffix))
+
         if ignore_uppercase_filenames and filename_without_extension.isupper():
             return
 
-        if new_name != filename_without_extension:
+        if messages:
             diff = Diff(file, rename=os.path.join(head, new_name + extension))
 
             yield Result(
                 self,
-                'Filename does not follow {} naming-convention.'.format(
-                    file_naming_convention),
+                '\n'.join('- ' + message for message in messages)
+                if len(messages) > 1 else messages[0],
```

Comment:
optional:
maybe put that into a `message` variable before, as this expression is a bit more complex?
up to you :+1: